sources/packages/api/src/esdc-integration/disbursement-receipt-integration/disbursement-receipt-processing.service.ts

```diff
@@ -0,0 +1,188 @@
+   * Once the file is processed, it gets deleted.
+   * @param auditUserId
+   * @returns Summary details of the processing.
+   */
+  async process(auditUserId: number): Promise<ProcessSFTPResponseResult> {
+    const result = new ProcessSFTPResponseResult();
+
+    // Get the list of all files from SFTP ordered by file name.
+    const fileSearch = new RegExp(
+      `^${this.esdcConfig.environmentCode}EDU.PBC.DIS.D[\w]*\.[0-9]*`,
+      "i",
+    );
+    const filePaths = await this.integrationService.getResponseFilesFullPath(
+      this.esdcConfig.ftpResponseFolder,
+      fileSearch,
+    );
+    if (filePaths.length === 0) {
+      result.processSummary.push(
+        "No files found to be processed at this time.",
+      );
+      return result;
+    }
+    for (const filePath of filePaths) {
+      await this.processAllReceiptsInFile(filePath, auditUserId, result);
+
+      try {
+        //Deleting the file once it has been processed.
+        await this.integrationService.deleteFile(filePath);
+      } catch (error) {
+        result.errorsSummary.push(
+          `Error while deleting disbursement receipt file: ${filePath}`,
+        );
+        result.errorsSummary.push(error);
+      }
+    }
+    return result;
+  }
+  /**
+   *
+   * @param remoteFilePath file which is to be processed.
+   * @param auditUserId user id of API user.
+   * @param result processing summary.
+   */
+  private async processAllReceiptsInFile(
+    remoteFilePath: string,
+    auditUserId: number,
+    result: ProcessSFTPResponseResult,
+  ): Promise<void> {
+    result.processSummary.push(`Processing file ${remoteFilePath}.`);
+    this.logger.log(`Starting download of file ${remoteFilePath}.`);
+    let responseData: DisbursementReceiptDownloadResponse;
+    try {
+      responseData = await this.integrationService.downloadResponseFile(
+        remoteFilePath,
+      );
+    } catch (error: unknown) {
+      this.logger.error(error);
+      result.errorsSummary.push(
+        `Error downloading file ${remoteFilePath}. Error: ${error}`,
+      );
+      return;
+    }
+
+    const documentNumbers = responseData.records.map(
+      (record) => record.documentNumber,
+    );
+    const disbursementSchedules =
+      await this.disbursementScheduleService.getDisbursementsByDocumentNumbers(
+        documentNumbers,
+      );
+    const disbursementScheduleMap = {};
+    disbursementSchedules.forEach(
+      (disbursementSchedule) =>
+        (disbursementScheduleMap[disbursementSchedule.documentNumber] =
+          disbursementSchedule.id),
+    );
+
+    this.logger.log(
+      "Processing all the records in file with valid data and document number that belongs to SIMS.",
+    );
+
+    const createdAt = new Date();
+    for (const response of responseData.records) {
+      const disbursementScheduleId =
+        disbursementScheduleMap[response.documentNumber];
+      if (disbursementScheduleId) {
```

Comment:
When the record is skipped because the document number is not present in our DB maybe it would be worth registering a log about it.